### PR TITLE
COST-195: Allow worker processing to scale on individual report file granularity

### PR DESCRIPTION
**Pull request by dccurtis**
This change breaks up the download/processing task monolith and makes the download/processing tasks work on an individual manifest file basis.

The orchestrator will process manifests and enqueue the appropriate report file processing celery tasks.  Once all report files are processed in a manifest the manifest summary is kicked off.

The orchestrator is also responsible now for checking for duplicate processing.

**Remaining Items:**
- [x] Code Cleanup
- [x] Manually test all providers and verify parquet functinonality
- [x] Re-work for parquet changes
- [x] Look into pod shutdowns while processor is running
- [X] Update unit tests to pass
- [x] Port Azure and GCP downloaders to allow for manifest downloads
- [x] Verify all temporary report files are removed.
- [x] Regression test Azure and OCP
- [x] Run smokes

**Functional Testing so far:**
1. Shutdown a worker while it's in the middle of processing.  Verify that summary does not run and the file that was in progress is not complete in the reporting_common_costusagereportstatus table. 
2. Hit the /download endpoint to simulate the next hourly polling processing period.  Verify that the last file in the manifest is processed and summary is complete.
3. Begin processing a manifest and hit the /download endpoint while processing is underway.  Verify that the second ingestion is blocked.
4. Ingest a manifest, generate and /download a new manifest while the first one is in progress.  Verify that first manifest completes processing and the new manifest will be processed at next /download period.
5. Stop the database while in the middle of processing.
6.  Ingest a large file count manifest where there's more files then workers.  Verify successful processing.
7. Sanity test OCP with ingress service
8. Sanity test Azure
9. Ingest AWS and OCP data and verify that parquet conversion is successful.
10. Stop a worker while in the middle of processing and verify the correct number of lines are processed when it restarts.

**Test Results:**
[worker_scaling_ut.txt](https://github.com/project-koku/koku/files/4859795/worker_scaling_ut.txt)

**Smokes Pass with 10 workers**
```
================================================================================================================================= 3454 passed, 80 skipped, 2186 deselected in 637.97s (0:10:37) ==================================================================================================================================
```

Deployment Config PR: https://github.com/RedHatInsights/e2e-deploy/pull/1982

### CODE REVIEW

File: koku/masu/processor/tasks.py [hunk 1]
```diff
@@ -107,69 +163,56 @@ def get_report_files(
...
-                    }
-                    reports_to_summarize.append(report_meta)
-            except (ReportProcessorError, ReportProcessorDBError) as processing_error:
-                worker_stats.PROCESS_REPORT_ERROR_COUNTER.labels(provider_type=provider_type).inc()
-                LOG.error(str(processing_error))
-                WorkerCache().remove_task_from_cache(cache_key)
-                raise processing_error
+    if report_dict:
+        stmt += " file: " + str(report_dict["file"]) + "\n"
+        LOG.info(stmt[:-1])
```
Comment (maskarb) [inline koku/masu/processor/tasks.py:168]:
```suggestion
        stmt += f" file: {report_dict['file']}"
        LOG.info(stmt)
```
this logging weirdness was necessary when we used the `for` loop.

File: koku/masu/processor/tasks.py [hunk 2]
```diff
@@ -107,69 +163,56 @@ def get_report_files(
...
+            f" provider: {provider_type}\n"
+            f" provider_uuid: {provider_uuid}\n"
+            f' file: {report_dict.get("file")}'
+        )
+        LOG.info(stmt)
+        worker_stats.PROCESS_REPORT_ATTEMPTS_COUNTER.labels(provider_type=provider_type).inc()
+        # import time
+
+        # LOG.info("Sleeping for 60 seconds")
+        # time.sleep(60)
```
Comment (maskarb) [inline koku/masu/processor/tasks.py:185]:
```suggestion
```

File: koku/masu/external/kafka_msg_handler.py [hunk 3]
```diff
@@ -567,9 +554,7 @@ def convert_manifest_to_parquet(request_id, report_meta, context={}):
                 "manifest_id": manifest_id,
                 "start_date": start_date,
             }
-            async_id = convert_reports_to_parquet.delay(
-                request_id=request_id, reports_to_convert=[report_meta], context=context
-            )
+
     return async_id
```
[the remark this answers is not in the document]
Comment (dccurtis) [reply koku/masu/external/kafka_msg_handler.py]: entire function was removed.  No longer needed

File: koku/masu/processor/orchestrator.py [hunk 4]
```diff
@@ -108,6 +116,90 @@ def get_reports(provider_uuid):
...
+        cache_key = f"{provider_uuid}:{month}"
+
+        manifest = None
+        downloader = ReportDownloader(
+            customer_name=customer_name,
+            access_credential=authentication,
+            report_source=billing_source,
+            provider_type=provider_type,
+            provider_uuid=provider_uuid,
+            cache_key=cache_key,
```
[the remark this answers is not in the document]
Comment (dccurtis) [reply koku/masu/processor/orchestrator.py]: no longer needed, removed

**Pull request merged**